vignettes/philosophy.Rmd

```diff
@@ -0,0 +1,163 @@
+
+### Rules for folder names
+
+- Folder names should only contain lower case letters, numbers and underscore (`_`).
+- They can start with a single dot (`.`).
+
+### Rules for file names
+
+- Base names should only contain lower case letters, numbers and underscore (`_`).
+- File extensions should only contains lower case letters and numbers.
+  Exceptions: file extensions related to R must have an upper case R (`.R`, `.Rmd`, `.Rd`, `.Rnw`, `.Rproj`).
+
+### Rules for graphical file names
+
+- Applies to files with extensions `csl`, `eps`, `jpg`, `jpeg`, `pdf`, `png` and `ps`.
+- Same rules except that you need to use a dash (`-`) as separator instead of an underscore (`_`).
+  We need to exception because underscores cause problems in certain situations.
+
+## Bundeling your code a package
+
+Most users think of an R package a collection of generic functions that they can use to run their analysis.
+However, an R package is useful way to bundle and document a stand alone analysis too!
+Suppose you want to pass your code to a collaborator or your future self who is working on a different computer.
+If you have a as project folder with a bunch of files, people will need to get to know your project structure, find out what scripts to run and which dependencies they need.
+Unless you documented everything well they (including your future self!) will a hard time figuring out how thing work.
+
+Having the analysis as a package _and_ running `check_package()` to ensure a minimal quality standard, make things a lot easier for the user.
+Agreed, it will take more time to create the analysis, especially with the first few projects.
+In the long run you save time due to a better quality of your code.
+
+### Benefits
+
+- The package itself is a way to cite (a specific version of) the analysis in a report or paper.
+- You have to list all dependencies on other R packages.
+  This makes installing your code as simple as running `remotes::install_github("inbo/packagename")`.
+- You must split your analysis in a set of functions.
+  Say goodbye to scripts with thousands lines of code.
+- Functions make it easy to re-use code.
+  Need to run the same thing with a different parameter values?
+  Add the parameter as an argument to the function and run the function once for every different parameter value.
+  This avoids the need to copy-paste large chunks of scripts and replace a few values.
+  The copy-paste work flow typically results in hard-to-read long scripts.
+  Image you made a mistake in the code and copy-pasted that mistake several times before you found it.
+  You have to check your entire project to fix the mistake.
+  Having it as a function reduces the workload to fixing only the function.
+- Packages require that every object is either defined within the package or imported from another package.
+  Global variables are not allowed.
+  This ensures that the functions inside your package only work with the objects defined inside the package.
```

Comment:
Is this what you actually meant to write?  Maybe it is even more clear if you just state that all functions can be used after a simple installation of the package (without additional objects needed).
```suggestion
  This ensures that the functions inside your package work with only the objects defined inside the package.
```